eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -20,20 +20,97 @@ import java.util.UUID
 
 import fr.acinq.bitcoin.ByteVector32
 import fr.acinq.eclair.MilliSatoshi
+import fr.acinq.eclair.crypto.Sphinx
+import fr.acinq.eclair.router.Hop
 
 import scala.compat.Platform
 
 /**
-  * Created by PM on 01/02/2017.
-  */
+ * Created by PM on 01/02/2017.
+ */
+
 sealed trait PaymentEvent {
   val paymentHash: ByteVector32
+  val timestamp: Long
+}
+
+case class PaymentSent(id: UUID, paymentHash: ByteVector32, paymentPreimage: ByteVector32, parts: Seq[PaymentSent.PartialPayment]) extends PaymentEvent {
+  require(parts.nonEmpty, "sent payment is empty")
+  val amount: MilliSatoshi = parts.map(_.amount).sum
+  val feesPaid: MilliSatoshi = parts.map(_.feesPaid).sum
+  val timestamp: Long = parts.map(_.timestamp).min
 }
 
-case class PaymentSent(id: UUID, amount: MilliSatoshi, feesPaid: MilliSatoshi, paymentHash: ByteVector32, paymentPreimage: ByteVector32, toChannelId: ByteVector32, timestamp: Long = Platform.currentTime) extends PaymentEvent
+object PaymentSent {
+
+  case class PartialPayment(id: UUID, amount: MilliSatoshi, feesPaid: MilliSatoshi, toChannelId: ByteVector32, route: Seq[Hop], timestamp: Long = Platform.currentTime)
```

Comment:
Actually you really need to provide a route all the time so I like having directly a Seq.
You're referring to the hackish part in the AuditDB where we set this to `Nil` because it's not really relevant for Audit right? I wanted to avoid creating yet another type, but I think putting an option there only because of this small AuditDB hack would be too harmful for the normal code...

Reply:
Not I was referring to the relayer